File: Assessment5/ImpurityIndex.java
class Impurity{
    public static int[] letter_occurence_computation(String sentence){
        char[] ch = sentence.toLowerCase().toCharArray();
        int[] letter_array = new int[26];
        for (int i=0 ; i< ch.length ; i++){
            try {
                letter_array[(int)ch[i] - 97] += 1;

            }
            catch(Exception e){
                continue;
            }
        }
        return letter_array;
    }

    public static int impurity_count(String sentence){
        int impurity_index = 0;
        int [] letter_array = Letter_occurence_computation(sentence);
        char [] vowels = {'a','e','i','o','u'};
        int [] impurity_increment = {0, 0.5, 0.7 ,1, 3};
        for (int i = 0; i < 26; i++){
            if(new String(vowels).indexOf(chr(i + 97)) != -1){
                if(letter_array[i] == 2) 
                    impurity_index += impurity_increment[1];
            }
            if(new String(vowels).indexOf(chr(i + 97)) == -1 && letter_array[i] == 2)
                impurity_index += impurity_increment[2];
            if(letter_array[i] == 3)
                impurity_index += impurity_increment[3];
            if(letter_array[i] > 3)
                impurity_index += impurity_increment[4];
        }
        return (impurity_index);
    }
    public static void main(String args[]){
        String s = "Old brother fox jumps over the lazy dog.";
        System.out.println(impurity_count(sentence));
    }
}

File: Assessment5/ImpurityIndex.py
def Letter_occurence_computation(sentence):
    letter_array = [0 for i in range(26)]
    for i in sentence.lower() :
        try :
            letter_array[ord(i) - 97] += 1
        except IndexError:
            pass
    return letter_array

def impurity_count(sentence):
    impurity_index = 0
    letter_array = Letter_occurence_computation(sentence)
    vowels = ['a','e','i','o','u']
    impurity_increment = [0, 0.5, 0.7 ,1, 3]
    for i in range(len(letter_array)):
        if(chr(i + 97) in vowels ):
            if(letter_array[i] == 2) :
                impurity_index += impurity_increment[1]
        if(chr(i + 97) not in vowels and  letter_array[i] == 2) :
            impurity_index += impurity_increment[2]
        if(letter_array[i] == 3):
            impurity_index += impurity_increment[3] 
        if(letter_array[i] > 3):
            impurity_index += impurity_increment[4] 
    return (impurity_index)

sentence = "Old brother fox jumps over the lazy dog."
print(impurity_count(sentence))
